### DISCUSSION TITLE
[Feature] Cross-User Access to Media Libraries with Customizable Permissions

### DISCUSSION THREAD

**Discussion by gabriellxxix** [immich-app/immich#14308] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

First, I would like to thank you for creating such an outstanding platform for managing and organizing media. Immich’s user-centric approach makes it an excellent tool for personal and collaborative use.

I would like to propose a feature that could significantly enhance the platform’s usability: the ability to share media (images and videos) between user accounts within the same instance, without duplicating the files.

Feature Description
The proposed feature would allow a user to make specific media items or folders visible and accessible to other user accounts. Permissions could be customized to fit different needs:

View-Only Access: The recipient can view the shared media without the ability to modify or delete it.
Collaborative Access: The recipient can view, modify metadata (e.g., tags or descriptions), and add comments to the shared media, but cannot delete it.
Full Access: The recipient can view, modify, and delete the shared media (ideal for trusted users like family members).
For example, a user could share all photos and videos taken from their phone with their spouse, allowing the spouse to view them directly within their own profile without needing to duplicate or download the files.

Benefits of This Feature
Enhanced Collaboration: Families, teams, or friends could easily share and manage media together.
Efficient Storage: By avoiding file duplication, users save significant storage space.
Personalized Experiences: Each user can maintain their own profile while still having access to shared content.
Suggested Implementation
Media Sharing Interface: Include an option to share specific items or folders from the media library, with a user-friendly permissions interface.
Shared Media Section: Add a section in the recipient's account where all shared media is grouped.
Customizable Permissions: Enable users to define the level of access when sharing media.
Audit Log (Optional): For collaborative or full-access permissions, maintain a log of changes made to shared media for transparency.
I believe this feature aligns with Immich’s mission of providing a seamless and intelligent media management solution. It would significantly enhance its value for users with collaborative needs, such as families or small teams.

Thank you for considering this suggestion. I would be happy to discuss this idea further or clarify any points if needed.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#12614
